tmt/utils.py

```diff
@@ -2656,11 +2656,21 @@ def default_branch(repository: Path, logger: tmt.log.Logger,
             return None
         if result.stdout is None:
             return None
-        git_parent_path = result.stdout.strip()
+        git_parent_path = result.stdout.strip().replace('/.git', '')
 
         # Should get the default branch of its parent
         work_repo = Path(git_parent_path)
 
+    # Make sure the '.git/refs/remotes/{remote}' directory is present
+    git_remotes_dir = work_repo / f'.git/refs/remotes/{remote}'
+    if not git_remotes_dir.exists():
+        return None
+
+    head = git_remotes_dir / 'HEAD'
+    # Make sure the HEAD reference is available
+    if not head.exists():
+        subprocess.run(f'git remote set-head {remote} --auto'.split(), cwd=work_repo)
```

Comment:
Why using `subprocess.run()` here instead of `run_command()` which is used for all other command executions?